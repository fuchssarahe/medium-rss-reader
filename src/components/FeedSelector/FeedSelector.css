.FeedSelector {
  text-align: center;
  display: flex;
  flex-direction: column;
  margin: auto;
  max-width: 15em;
  justify-content: space-around;
}

.FeedSelector__input {
  padding: 0.5em;
  margin-top: 1em;
  font-size: 16px;
}

 .FeedSelector__button {
   padding: 0.5em;
   margin-top: 1em;
   border-radius: 0.5em;
   font-size: 16px;
 }
