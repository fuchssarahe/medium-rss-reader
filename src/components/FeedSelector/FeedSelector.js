import React, { Component } from 'react';
import './FeedSelector.css';

class FeedSelector extends Component {
  constructor(props) {
    super(props);
    this.state = { feedName: null };
  }

  handleSubmit = (event) => {
    event.preventDefault();
    this.props.handleSubmit(this.state.feedName);
  }

  handleChange = (event) => {
    this.setState({ feedName: event.target.value })
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit} className='FeedSelector'>
        <input type="text" name="feed-name" onChange={this.handleChange} className='FeedSelector__input'/>
        <button type="submit" className='FeedSelector__button'>View Feed</button>
      </form>
    );
  }
}

export default FeedSelector;
