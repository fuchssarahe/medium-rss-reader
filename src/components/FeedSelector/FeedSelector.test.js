import React from 'react';
import { shallow, mount } from 'enzyme';
import FeedSelector from './FeedSelector';

describe('<FeedSelector />', () => {
  test('it renders a form with input and submit button', () => {
    expect(shallow(
      <FeedSelector handleSubmit={ () => {} } />
    )).toMatchSnapshot();
  });

  test('it submits the form with the user-entered data', () => {
    const actionSpy = jest.fn();

    let component = mount(<FeedSelector handleSubmit={actionSpy} />);
    component
      .find('.FeedSelector__input')
      .first()
      .simulate('change', { target: { value: 'go' } })
    component.find('.FeedSelector').first().simulate('submit');

    expect(actionSpy).toHaveBeenCalledWith('go');
    component.unmount();
  });
});
