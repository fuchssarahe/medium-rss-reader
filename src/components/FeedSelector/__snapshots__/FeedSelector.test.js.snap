// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedSelector /> it renders a form with input and submit button 1`] = `
<form
  className="FeedSelector"
  onSubmit={[Function]}
>
  <input
    className="FeedSelector__input"
    name="feed-name"
    onChange={[Function]}
    type="text"
  />
  <button
    className="FeedSelector__button"
    type="submit"
  >
    View Feed
  </button>
</form>
`;
